src/lib/hooks/useApproval.ts

```diff
@@ -14,29 +14,41 @@ export enum ApprovalState {
   APPROVED = 'APPROVED',
 }
 
+export interface ApprovalStateForSpenderParams {
+  approvalState: ApprovalState
+}
+
 export function useApprovalStateForSpender(
   amountToApprove: CurrencyAmount<Currency> | undefined,
   spender: string | undefined,
   useIsPendingApproval: (token?: Token, spender?: string) => boolean
-): ApprovalState {
+): ApprovalStateForSpenderParams {
   const { account } = useWalletInfo()
   const token = amountToApprove?.currency?.isToken ? amountToApprove.currency : undefined
 
   const currentAllowance = useTokenAllowance(token, account ?? undefined, spender)
   const pendingApproval = useIsPendingApproval(token, spender)
 
   return useMemo(() => {
-    if (!amountToApprove || !spender) return ApprovalState.UNKNOWN
-    if (amountToApprove.currency.isNative) return ApprovalState.APPROVED
+    if (!amountToApprove || !spender) return { approvalState: ApprovalState.UNKNOWN }
+    if (amountToApprove.currency.isNative) return { approvalState: ApprovalState.APPROVED }
     // we might not have enough data to know whether or not we need to approve
-    if (!currentAllowance) return ApprovalState.UNKNOWN
+    if (!currentAllowance)
+      return {
+        approvalState: ApprovalState.UNKNOWN,
+      }
 
     // amountToApprove will be defined if currentAllowance is
-    return currentAllowance.lessThan(amountToApprove)
+    const approvalState = currentAllowance.lessThan(amountToApprove)
       ? pendingApproval
         ? ApprovalState.PENDING
         : ApprovalState.NOT_APPROVED
       : ApprovalState.APPROVED
+
+    return {
+      approvalState,
+      currentAllowance,
```

Comment:
Wait... `currentAllowance` is not part of the return type.
Doesn't TS complain about this?

Reply:
it didn't :) 

I meant to add it